### PR TITLE
Privacy policy

### PR DESCRIPTION
**Pull request by Chrisbh**
This is the backend of the simple implementation of privacy policy, a more advanced version will come later.

See the issue here:  https://github.com/Energinet-DataHub/energy-origin/issues/289

Work done here:
- Moved several functions and other bits of code into a different file to make it easier to reuse
- Removed some of the SSN implementation which was getting in the way, we can always go back in the source control and find it again when needed
- Added endpoints for get and accept terms
- Added a create user endpoint
- Tests for all the logic and the flow

Other small things added
- Flake8 linting enabled for vscode
- Fixed an issue with devcontainers

### CODE REVIEW

File: src/auth_api/config.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 from decouple import config
-
+import os
```
Comment (j4k0bk) [inline src/auth_api/config.py:2]: Flyt denne linje op øverst i filen

File: src/auth_api/config.py [hunk 2]
```diff
@@ -18,6 +18,13 @@
 # Service' public URL
 SERVICE_URL = config('SERVICE_URL', default=DEVELOP_URL)
 
+# Source path
+SOURCE_DIR = os.path.dirname(os.path.abspath(__file__))
+ROOT_DIR = os.path.join(SOURCE_DIR, '..')
+TERMS_MARKDOWN_PATH = os.path.join(ROOT_DIR, 'terms', 'terms.md')
+
+# No. of hours used for the timedelta for internal token expiry
+TOKEN_EXPIRY_DELTA = 1
```
Comment (j4k0bk) [inline src/auth_api/config.py:27]: Lav det her udtryk om til en timedelta, eks:

from datetime import timedelta

TOKEN_EXPIRY_DELTA = timedelta(hours=1)

Comment (j4k0bk) [inline src/auth_api/config.py:27]: Bruger kan max. være logget ind i en time

File: src/auth_api/endpoints/terms.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
+from dataclasses import dataclass
+
+import markdown2
```
Comment (j4k0bk) [inline src/auth_api/endpoints/terms.py:3]: Flyt denne linje op øverst i filen

File: src/auth_api/endpoints/terms.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        headline: str
+        terms: str
+        version: str
+
+    def handle_request(self, context: Context) -> Response:
+        """
+        Handle HTTP request.
+        """
+        with open(TERMS_MARKDOWN_PATH) as f:
+            markdown_to_html = markdown2.markdown(f.read())
```
Comment (j4k0bk) [inline src/auth_api/endpoints/terms.py:29]: Måske bare kald variablen "html" eller "markdown_html" ?

File: src/auth_api/endpoints/oidc.py [hunk 5]
```diff
@@ -383,53 +301,55 @@ def on_oidc_flow_succeeded(
...
-                identity_provider=token.provider,
-                external_subject=token.subject,
+        if not state.terms_accepted:
+            return redirect_to_failure(
+                state=state,
+                error_code='E4',
             )
 
-        return super(OpenIDSsnCallback, self).on_oidc_flow_succeeded(
+        user = db_controller.get_or_create_user(
```
Comment (j4k0bk) [inline src/auth_api/endpoints/oidc.py:336]: Vi skal også gemme, hvilken version af terms, brugeren har accepteret.
Der skal tilføjes en attribut på DbUser og laves en database migration.

Comment (Chrisbh) [reply src/auth_api/endpoints/oidc.py:336]: Det kommer som en del af næste opgave ift. versionering.

File: src/auth_api/endpoints/oidc.py [hunk 6]
```diff
@@ -165,29 +146,40 @@ def handle_request(
...
             )
         except Exception:
             # TODO Log this exception
-            return self._redirect_to_failure(
+            return redirect_to_failure(
                 state=state,
                 error_code='E505',
             )
 
+        if oidc_token.expires < datetime.now(tz=timezone.utc):
```
Comment (j4k0bk) [inline src/auth_api/endpoints/oidc.py:161]: Der mangler en test case af dette (hvis du synes det er relevant)

File: src/auth_api/endpoints/oidc.py [hunk 7]
```diff
@@ -165,29 +146,40 @@ def handle_request(
...
+            return redirect_to_failure(
+                state=state,
+                error_code='E505',
+            )
+
+        state.tin = oidc_token.tin
```
Comment (j4k0bk) [inline src/auth_api/endpoints/oidc.py:167]: Der er så vidt jeg kan se ikke nogen test-case, der tjekker, at disse værdier sættes korrekt på AuthState

**Review by j4k0bk — APPROVED** [review]

**Review by j4k0bk — APPROVED** [review]

**Pull request merged**